tools/distribution/distrotest.py

```diff
@@ -0,0 +1,715 @@
+            path: pathlib.Path,
+            tarball: pathlib.Path,
+            keyfile: pathlib.Path,
+            testfile: pathlib.Path,
+            maintainer: str,
+            version: str,
+            config_path: Optional[pathlib.Path] = None):
+        self.docker = docker
+        self._keyfile = keyfile
+        self.path = path
+        self.tarball = tarball
+        self._testfile = testfile
+        self.maintainer = maintainer
+        self.version = version
+        self._config_path = config_path
+
+    def __getitem__(self, k):
+        return self.config[k]
+
+    @cached_property
+    def config(self) -> dict:
+        """Configuration for test types - eg deb/rpm
+
+        This contains build information for different types of test image,
+        and some defaults for specific test configuration.
+        """
+        return utils.from_yaml(self.config_path)
+
+    @cached_property
+    def config_path(self) -> pathlib.Path:
+        """Path to a test configuration file"""
+        return pathlib.Path(self._config_path or DISTROTEST_CONFIG_PATH)
+
+    @cached_property
+    def ctx_dockerfile(self) -> pathlib.Path:
+        """Path to the Dockerfile in the Docker context"""
+        return self.path.joinpath("Dockerfile")
+
+    @cached_property
+    def ctx_keyfile(self) -> pathlib.Path:
+        """Path to the keyfile in the Docker context"""
+        return self.path.joinpath(self._keyfile.name)
+
+    @cached_property
+    def ctx_packages(self) -> pathlib.Path:
+        """Path to the directory (in the Docker context) containing packages to
+        test
+        """
+        return self.path.joinpath(self.packages_name)
+
+    @cached_property
+    def ctx_testfile(self) -> pathlib.Path:
+        """Path to the testfile in the Docker context"""
+        return self.path.joinpath(self._testfile.name)
+
+    @cached_property
+    def images(self) -> dict:
+        """Mapping of images -> ext/types
+
+        eg `debian` -> type=`deb` ext=`changes`
+           `registry.access.redhat.com/ubi8/ubi` -> type=`rpm` ext=`rpm`
+
+        for each image:
+          - the `type` is used to find the directory of packages.
+          - the `ext` is used to find packages within the directory.
+
+        """
+        return dict(
+            chain.from_iterable(((image, dict(type=k, ext=v["ext"]))
+                                 for image in v["images"])
+                                for k, v in self.items()))
+
+    @cached_property
+    def install_dir(self) -> str:
+        """Path to the install directory within the image/container"""
+        return "/tmp/install"
+
+    @cached_property
+    def key_path(self) -> str:
+        """Path to the public key of the key used to sign the packages, inside
+        the Docker image/container
+        """
+        return "/tmp/gpg/signing.key"
+
+    @cached_property
+    def keyfile(self) -> pathlib.Path:
+        """Path to the keyfile in the Docker context
```

Comment:
OCD nit: I don't see the rationale that makes `key_path` have a `_path` suffix but `keyfile` have a `file` one. Would really be nice to have consistent terminiology.

Reply:
well the nomenclature is maybe not great but there is a rationale

`key_path` is a path inside the image/container

related `ctx_foo` are paths in the folder which gets dockerized

Reply:
checking further i think i have been consistent in use of `_path` to denote "inside the image" with the exception of `config_path` which is just the path to the config file (ie on the host)

Reply:
Yeah, maybe make more explicit, e.g. `key_imagepath`, `key_hostpath`, whatever would be most clear.

Reply:
k - ill update...

Reply:
updated so  that all properties and methods that provide a path inside the image/container use `PurePosixPath` and end with `_img_path`